### PR TITLE
Add facerec path to config and modify code accordingly

### PR DESCRIPTION
**Pull request by pulkitvyas08**
Solves #20

### CODE REVIEW

**Review by rushabh-v — CHANGES_REQUESTED** [review]
Change the installation paths under `debian/` as well (wherever required)

**Review by rushabh-v — CHANGES_REQUESTED** [review]
Do not push `.pyc` files

File: src/utils/add_new.py [hunk 1]
```diff
@@ -1,14 +1,20 @@
-from os import system
+from os import system, path
 
-import getFaces
+import sys, inspect
+current_dir = path.dirname(path.abspath(inspect.getfile(inspect.currentframe())))
+src_dir = path.dirname(current_dir)
+main_dir = path.dirname(src_dir)
+sys.path.insert(0, main_dir)
```
Comment (rushabh-v) [inline src/utils/add_new.py]: Why is this required?

Comment (pulkitvyas08) [reply src/utils/add_new.py]: To import config from 2 directories upward. But I found a better way now. Will push it with the next commit shortly

Comment (rushabh-v) [reply src/utils/add_new.py]: After installation, everything will be under the same directory(`/usr/lib/Auth/Facerec/`). So, that will not be required.

Comment (pulkitvyas08) [reply src/utils/add_new.py]: Oh yes you're right. That's why I've been getting failed tests. Thanks!

**Comment by rushabh-v** [comment]
Don't change the directory for `pam_python.so`. That should be under `/lib/x86_64-linux-gnu/security/` only. Also change the paths wherever required in `Facerec`, `src/cli/..`, etc.

**Comment by rushabh-v** [comment]
About the currently failing test, try importing `.config` everywhere.

**Comment by pulkitvyas08** [comment]
> About the currently failing test, try importing `.config` everywhere.

Sorry but what exactly do you mean by that?

File: src/utils/add_new.py [hunk 2]
```diff
@@ -1,14 +1,14 @@
-from os import system
+from os import system, path
+from config import PATH
```
Comment (rushabh-v) [inline src/utils/add_new.py]:
```suggestion
from .config import PATH
```

File: src/utils/compare.py [hunk 3]
```diff
@@ -7,12 +8,14 @@
  np,
 )
 
+from config import PATH
```
Comment (rushabh-v) [inline src/utils/compare.py]:
```suggestion
from .config import PATH
```

**Comment by pulkitvyas08** [comment]
Okay got it

File: src/utils/getFaces.py [hunk 4]
```diff
@@ -1,7 +1,10 @@
 import sys
 import numpy as np
+from os import path
 
-deps_path = list(np.load('/lib/Auth/Facerec/deps_path.npy'))
+from config import PATH
```
Comment (rushabh-v) [inline src/utils/getFaces.py]:
```suggestion
from .config import PATH
```

**Comment by rushabh-v** [comment]
I have updated it. Pull `origin/master` to your local branch before making further changes.

**Comment by pulkitvyas08** [comment]
My merge commit also got in the pull request. Will take care of that from next time

**Comment by rushabh-v** [comment]
No, that's fine!

**Comment by rushabh-v** [comment]
Remove the `.pyc` files once it passes the tests. I'll try to install it locally. And will merge once everything works fine!

File: src/utils/add_new.py [hunk 5]
```diff
@@ -1,14 +1,14 @@
-from os import system
+from os import system, path
+from .config import PATH
```
Comment (rushabh-v) [inline src/utils/add_new.py:2]:
```suggestion
from config import PATH
```

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -22,8 +22,8 @@ jobs:
 
     - name: Check imports
       run: |
-        python3 /lib/Auth/Facerec/getFaces.py
-        python /lib/Auth/Facerec/pam_ptn.py
+        python3 /usr/lib/Auth/Facerec/getFaces.py
+        python /usr/lib/Auth/Facerec/pam_ptn.py
```
Comment (rushabh-v) [inline .github/workflows/main.yml]:
```suggestion
        sudo python3 /usr/lib/Auth/Facerec/getFaces.py
        sudo python /usr/lib/Auth/Facerec/pam_ptn.py
```

File: src/utils/add_new.py [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
 import getFaces
 
-
 CRED = '\33[91m'
 CGRN = '\33[42m'
 CEND = '\33[0m'
 
 if __name__ == '__main__':
-    path = '/lib/Auth/Facerec/roots/'
+    path = path.join(config.PATH,'roots/')
```
Comment (rushabh-v) [inline src/utils/add_new.py]:
```suggestion
    path = path.join(config.PATH, 'roots/')
```

**Comment by rushabh-v** [comment]
Will test it locally by tomorrow. Will merge if everything works fine.

File: src/utils/add_new.py [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
 import getFaces
 
-
 CRED = '\33[91m'
 CGRN = '\33[42m'
 CEND = '\33[0m'
 
 if __name__ == '__main__':
-    path = '/lib/Auth/Facerec/roots/'
+    path = path.join(config.PATH, 'roots/')
```
Comment (rushabh-v) [inline src/utils/add_new.py]: Gives `UnboundLocalError: local variable 'path' referenced before assignment` on this line. Let's keep it simple string concatenation for this PR (it will reduce testing efforts). And will use `pathlib` (`pathlib` is better when it comes to working with paths.) to replace those string paths with `pathlib.Path` in another PR as it's not in the scope of this PR.

**Review by rushabh-v — CHANGES_REQUESTED** [review]

File: src/utils/add_new.py (hunk 8 above)
Comment (pulkitvyas08) [reply src/utils/add_new.py]: Okay then, for now I'll replace all the path.join with string concatenation

**Comment by pulkitvyas08** [comment]
Ignore the reverts, I made some changes to the workflow for a different PR but forgot to undo them before pushing

**Comment by rushabh-v** [comment]
Works fine. Good job @pulkitvyas08

**Comment by rushabh-v** [comment]
give @pulkitvyas08 75 points
